### PR TITLE
Storybook: add args table to `BarChart`

### PR DESCRIPTION
**Pull request by Yilun-Sun**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: storybook/stories/API/chart/BarChart.mdx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+# BarChart
+
+## Parent Component
+
+The BarChart can be used within: `<ResponsiveContainer />`.
+
+## Child component
+
+The BarChart can be used with the following child components: `<XAxis />`, `<YAxis />`, `<ReferenceArea />`, `<ReferenceDot />`, `<ReferenceLine />`, `<Brush />`, `<CartesianGrid />`, `<Legend />`, `<Tooltip />`, `<Bar />`, `<Customized />` or validate svg elements.
```
Comment (ckifer) [inline storybook/stories/API/chart/BarChart.mdx:12]: this was probably in the old docs but 'validate' should be 'valid'

File: storybook/stories/API/chart/BarChart.stories.tsx [hunk 2]
```diff
@@ -1,9 +1,188 @@
...
+    table: {
+      type: {
+        summary: "'horizontal' | 'vertical'",
+        defaultValue: "'horizontal'",
+      },
+      category: 'General',
+    },
+  },
+  syncId: {
+    description: `If any two categorical charts(LineChart, AreaChart, BarChart, ComposedChart) have the same syncId,
```
Comment (ckifer) [inline storybook/stories/API/chart/BarChart.stories.tsx:18]: nit nit: can we add a space between charts and the parenthesis? "categorical charts (LineChart..."

**Review by ckifer — APPROVED** [review]
LGTM but would prefer to fix the nits before merging. Thank you as always @Yilun-Sun 🙏🏼

File: storybook/stories/API/chart/BarChart.stories.tsx (hunk 2 above)
Comment (nikolasrieble) [inline storybook/stories/API/chart/BarChart.stories.tsx:18]:
```suggestion
    description: `If any two categorical charts (LineChart, AreaChart, BarChart, ComposedChart) have the same syncId, 
```

File: storybook/stories/API/chart/BarChart.mdx (hunk 1 above)
Comment (nikolasrieble) [inline storybook/stories/API/chart/BarChart.mdx:12]:
```suggestion
The BarChart can be used with the following child components: `<XAxis />`, `<YAxis />`, `<ReferenceArea />`, `<ReferenceDot />`, `<ReferenceLine />`, `<Brush />`, `<CartesianGrid />`, `<Legend />`, `<Tooltip />`, `<Bar />`, `<Customized />` or valid svg elements.
```

**Review by nikolasrieble — APPROVED** [review]
Great, thank you! Lets iterate and remove redundant descriptions in a follow up.

**Pull request merged**